### ISSUE TITLE
Use shared informer factory's WaitForCacheSync in server.go

### ISSUE THREAD

**Issue by ncdc** [heptio/velero#616]
The shared informer factory has a `WaitForCacheSync` method that we can call that will wait for all started informers' caches to sync. For all the places where we have code individually calling `cache.WaitForCacheSync()` with a list of informers **from our standard shared informer factories**, we should be able to remove these calls and rely on a single `factory.WaitForCacheSync()` call instead, in `server.go`.

We also have some informers that are created by hand, separate from the shared informer factories. For these informers, we still need to call `cache.WaitForCacheSync` separately.

**Comment by skriss** [comment]
What exactly would this flow look like?  Things at play:

- instantiating informers (from the factory)
- adding event handlers (in controllers' constructors)
- starting the informers (in `server.go`)
- waiting for cache sync (currently in controllers, before processing items)

Specifically wondering about how to coordinate adding event handlers, then starting & waiting for cache sync, before having controllers process items.

**Comment by ncdc** [comment]
1. Create factory
1. Create all controllers (they add event handlers)
1. Start factory
1. Wait for factory cache sync
1. Run all controllers

I now right now we combine creating and running each controller, and do that for all controllers, prior to starting the shared informer factories. We should be able to use the steps above - the key requirement is that the event handlers are added before we start the factories.

**Comment by ncdc** [comment]
This is low priority and not something I've tried before, but it could make adding in new informers or using new informers in existing controllers easier.

**Comment by skriss** [comment]
Thanks for this request. We are not currently prioritizing this item within our roadmap, and as such will be closing out the issue. Please feel free to reach out as needed in this issue or a new one.
